pkg/languagedetection/detector.go

```diff
@@ -36,34 +36,42 @@ type languageFromCLI struct {
 	validator func(exe string) bool
 }
 
-// rubyPattern is a regexp validator for the ruby prefix
-var rubyPattern = regexp.MustCompile(`^ruby\d+\.\d+$`)
+var (
+	rubyPattern = regexp.MustCompile(`^ruby\d+\.\d+$`)
+	phpPattern  = regexp.MustCompile(`^ruby\d$`)
```

Comment:
```suggestion
	phpPattern  = regexp.MustCompile(`^php\d$`)
```
Unless naming is wrong!

Reply:
Unsure how this regex matches `php index.php`. It seems to match only `php5 index.php`. Also, you might want to add the same for `php-fpm` and `php-fpm<major>`.

Reply:
@bwoebi any other pattern that comes to your mind we should detect?

Reply:
It also looks like the test is passing because it's falling through looking at the processes's `comm` value [here](https://github.com/DataDog/datadog-agent/blob/4bf67b8b9188d16440bb1334b6f05c7a59b90d95/pkg/languagedetection/detector.go#L140-L142)

This is `php` which matches the known prefixes.

Reply:
If the goal of that function is to avoid running the regex _only_ if strictly necessary because of performance reasons, then the pattern with the required `\d` is fine and the additional work would only be to include the pretty ubiquitous `php-fpm` case.

Reply:
Thanks for catching that!